lib/puppet/util/windows/process.rb

```diff
@@ -224,6 +224,56 @@ def windows_major_version
   end
   module_function :windows_major_version
 
+  ENVSTRINGS_TERMINATOR_WCHAR = [0,0]
+  # Returns a hash of the current environment variables encoded as UTF-8
```

Comment:
Looks like because we're calling `read_arbitrary_wide_string_up_to`, we're calling `read_wide_string`, which re-encodes the UTF-16LE string as UTF-8.  :+1: